src/docs/timestamp-global.dox

```diff
@@ -144,33 +126,39 @@ or \c oldest_reader if there are no active transactions.
 
 @subsection timestamp_global_query_api_all_durable Reading the "all_durable" timestamp
 
-<!--
-    XXX Given the constraints on transactional durable timestamps, it
-    seems like all_durable must always be >= stable, no?
-
-    Also, this could use clarification. My understanding from what's
-    here is that it's a scheme for tracking what has committed
-    recently, and has no direct relationship to what's checkpointed on
-    disk and thus _actually_ durable (that's last_checkpoint)... but I
-    am not clear on what one does with it if you need to have your own
-    application-level locks in order to read it usefully. A description
-    of how to use it would be helpful, but I don't know what to write.
-    Also, if there are no running transactions, does it reflect the
-    durable timestamps of committed transactions, reset to 0, or
-    something else?
--->
-
 The \c all_durable timestamp is the minimum of the global \c durable_timestamp
-(as set by the application), and the durable timestamps of all currently running
-transactions in the system.
+value and a durable timestamp before any unresolved transaction in the system.
+
+For example:
+- If a transaction commits with a durable timestamp of 50 and there are no
+running transactions in the system, \c all_durable will return 50.
+- If a transaction commits with a durable timestamp of 50, and there is
+a running transaction with a durable timestamp of 40, \c all_durable will
+return 39.
+- If a transaction commits with a durable timestamp of 50 and there
+are no running transactions in the system and the application sets
+\c durable_timestamp to 30, \c all_durable will return 30.
+- If a transaction commits with a durable timestamp of 50, and there is a
+running transaction with a durable timestamp of 20 and the application sets
+\c durable_timestamp to 30, \c all_durable will return 19.
+- If the application sets \c durable_timestamp to 30, and a transaction then
+commits with a durable timestamp of 50, \c all_durable will return 50.
+
+\warning It is possible for the \c all_durable value read by the application
+to be obsolete by the time the application receives it and applications using
+this value to make operational decisions are responsible for coordinating their
+own actions (for example by using high-level locks) to ensure its validity.
+Specifically, any transaction commit with a durable timestamp larger than than
+the current \c durable_timestamp value will set the \c durable_timestamp to the
+transaction's durable timestamp. Calling the WT_CONNECTION::rollback_to_stable
+method will reset the \c durable_timestamp to the stable timestamp.
```

Comment:
This explains a number of things that were previously obscure, but there's still one point I don't really understand: most running transactions don't have a durable timestamp because the durable timestamp isn't set until either commit or right before commit. I am guessing that in order for this to be useful, MDB sets the commit and durable timestamps first and then waits for distributed stuff to resolve? Otherwise it seems to be just about which transactions are in the middle of committing.

Anyway it might be helpful to explicitly say that running transactions that haven't had a durable timestamp assigned yet (which is the last step before committing) don't affect the result. Assuming that's true.

Reply:
If no durable timestamp is set, the transaction's commit timestamp is used to advance the system's "durable" timestamp. Fixed in 11d6806.

Reply:
That's a good point, but a different one -- what I was wondering about is a transaction that's "running" but hasn't had any timestamp set yet.

Reply:
(This has been sorted out offline.)